chronos/asyncmacro2.nim

```diff
@@ -81,11 +83,19 @@ template useVar(result: var NimNode, futureVarNode: NimNode, valueReceiver,
 
 template createVar(result: var NimNode, futSymName: string,
                    asyncProc: NimNode,
-                   valueReceiver, rootReceiver: untyped,
+                   valueReceiver, rootReceiver, retFutSym: untyped,
                    fromNode: NimNode) =
   result = newNimNode(nnkStmtList, fromNode)
   var futSym = genSym(nskVar, "future")
   result.add newVarStmt(futSym, asyncProc) # -> var future<x> = y
+  # retFuture.child = future<x>
```

Comment:
I think all of this template can greatly benefit from using `quote do` notation

Reply:
I even do not understand why this templates not compile-time procedures?